onnxruntime/python/tools/transformers/notebooks/Inference_GPT2-OneStepSearch_OnnxRuntime_CPU.ipynb

```diff
@@ -0,0 +1,544 @@
+   ],
+   "source": [
+    "tokenizer = get_tokenizer(model_name_or_path, cache_dir)\n",
+    "input_text = EXAMPLE_Text\n",
+    "test_generation(tokenizer, input_text, use_onnxruntime_io=False, ort_session=session)"
+   ]
+  },
+  {
+   "source": [
+    "Next, we use ONNX Runtime with IO binding to run again and we can see that the result is exactly same."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "Text generation using OnnxRuntime with IO binding ...\n",
+      "------------\n",
+      "best hotel in bay area.\n",
+      "\n",
+      "\"It's a great place to stay,\" he said. \"It's a great place to live. It's a great place to work\n"
+     ]
+    }
+   ],
+   "source": [
+    "test_generation(tokenizer, input_text, use_onnxruntime_io=True, ort_session=session)"
+   ]
+  },
+  {
+   "source": [
+    "## Int8 Quantization ##\n",
+    "Next, we will apply dynamic quantization to the model. We optimize the model before quantization to get better performance.\n",
+    "\n",
+    "Note that text generation result from fp32 and int8 models could be quite different. User shall evaluate the precision metric for your application for both fp32 and int8 models. If the quality of int8 model result is acceptable, you will be glad to find that it is faster than fp32 model in inference. \n",
+    "\n",
+    "Note that you can leverage [quantization aware training (QAT)](https://pytorch.org/blog/introduction-to-quantization-on-pytorch/) for accuracy improvement if needed."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stderr",
+     "text": [
+      "WARNING:root:onnxruntime.quantization.quantize is deprecated.\n         Please use quantize_static for static quantization, quantize_dynamic for dynamic quantization.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from onnxruntime.transformers.quantize_helper import QuantizeHelper\n",
+    "\n",
+    "optimized_fp32_model_path = \"gpt2_one_step_search_fp32.onnx\"\n",
+    "quantized_int8_model_path = \"gpt2_one_step_search_int8.onnx\"\n",
+    "Gpt2BeamSearchHelper.optimize_onnx(\"gpt2_one_step_search.onnx\", optimized_fp32_model_path, False, model.config.num_attention_heads, model.config.hidden_size)\n",
+    "QuantizeHelper.quantize_onnx_model(optimized_fp32_model_path, quantized_int8_model_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "Text generation using OnnxRuntime ...\n",
+      "------------\n",
+      "GPT-2 model optimization with one step search.\n",
+      "\n",
+      "SUMMARY OF THE INVITATION\n",
```

Comment:
It seems that int8 model result is quite different. Any idea that whether it is caused by quantization or something wrong?

Reply:
I have removed this section for now. I will do some further analysis on quantizing one step search module